edx_exams/apps/core/tests/test_api.py

```diff
@@ -289,6 +290,102 @@ def test_with_no_attempt(self):
         self.assertIsNone(get_attempt_by_id(111111111))
 
 
+@ddt.ddt
+class TestGetAttemptsInProgress(ExamsAPITestCase):
+    """
+    Test for the API utility function `get_latest_attempt_for_user`
+    """
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = '11111111'
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+        self.user = User.objects.create(
+            id=3,
+            username="jerry",
+            email="jerry@example.com",
+            lms_user_id=2
+        )
+
+        # Create mock exam attempt data
+        self.expected_attempt = ExamAttempt.objects.create(  # This attempt is what we expect the get function to return
```

Comment:
NIT: we're typing out the same code over and over here. Since this is just creating a bunch of objects with similar attributes I'd recommend creating a helper function to do this.

```
create_exam_attempt(self, status, start_time=None):
    return ExamAttempt.objects.create(
            user=self.user,
            ...
```

You can actually always make the attempt number 1 here to make things easier. That value is an external key that isn't used by this endpoint.